crates/bevy_ecs/src/query/access.rs

```diff
@@ -1,67 +1,30 @@
 use crate::component::ComponentId;
+use crate::storage::SortedVecSet;
 use crate::storage::SparseSetIndex;
 use crate::world::World;
 use alloc::{format, string::String, vec, vec::Vec};
-use core::{fmt, fmt::Debug, marker::PhantomData};
+use core::{fmt::Debug, marker::PhantomData};
 use derive_more::derive::From;
 use disqualified::ShortName;
-use fixedbitset::FixedBitSet;
 
-/// A wrapper struct to make Debug representations of [`FixedBitSet`] easier
-/// to read, when used to store [`SparseSetIndex`].
-///
-/// Instead of the raw integer representation of the `FixedBitSet`, the list of
-/// `T` valid for [`SparseSetIndex`] is shown.
-///
-/// Normal `FixedBitSet` `Debug` output:
-/// ```text
-/// read_and_writes: FixedBitSet { data: [ 160 ], length: 8 }
-/// ```
-///
-/// Which, unless you are a computer, doesn't help much understand what's in
-/// the set. With `FormattedBitSet`, we convert the present set entries into
-/// what they stand for, it is much clearer what is going on:
-/// ```text
-/// read_and_writes: [ ComponentId(5), ComponentId(7) ]
-/// ```
-struct FormattedBitSet<'a, T: SparseSetIndex> {
-    bit_set: &'a FixedBitSet,
-    _marker: PhantomData<T>,
-}
-
-impl<'a, T: SparseSetIndex> FormattedBitSet<'a, T> {
-    fn new(bit_set: &'a FixedBitSet) -> Self {
-        Self {
-            bit_set,
-            _marker: PhantomData,
-        }
-    }
-}
-
-impl<'a, T: SparseSetIndex + Debug> Debug for FormattedBitSet<'a, T> {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_list()
-            .entries(self.bit_set.ones().map(T::get_sparse_set_index))
-            .finish()
-    }
-}
+const ACCESS_SMALL_VEC_SIZE: usize = 8;
 
 /// Tracks read and write access to specific elements in a collection.
 ///
 /// Used internally to ensure soundness during system initialization and execution.
 /// See the [`is_compatible`](Access::is_compatible) and [`get_conflicts`](Access::get_conflicts) functions.
-#[derive(Eq, PartialEq)]
+#[derive(Debug, Eq, PartialEq)]
 pub struct Access<T: SparseSetIndex> {
     /// All accessed components, or forbidden components if
     /// `Self::component_read_and_writes_inverted` is set.
-    component_read_and_writes: FixedBitSet,
+    component_read_and_writes: SortedVecSet<ACCESS_SMALL_VEC_SIZE>,
     /// All exclusively-accessed components, or components that may not be
     /// exclusively accessed if `Self::component_writes_inverted` is set.
-    component_writes: FixedBitSet,
+    component_writes: SortedVecSet<ACCESS_SMALL_VEC_SIZE>,
     /// All accessed resources.
-    resource_read_and_writes: FixedBitSet,
+    resource_read_and_writes: SortedVecSet<ACCESS_SMALL_VEC_SIZE>,
     /// The exclusively-accessed resources.
-    resource_writes: FixedBitSet,
+    resource_writes: SortedVecSet<ACCESS_SMALL_VEC_SIZE>,
```

Comment:
Is it at all feasible, within this PR or a follow up one, to factor the `x` and `x_inverted` pairings into a utility struct that wraps the `SortedVecSet` and `inverted` boolean, encapsulating all of the logic for inverting these sets? 😮

Reply:
I vote for doing that as a follow-up!  It would be a good simplification, but it's not related to the goal of this PR and I don't think we should try to do too many things at once.